src/lib/executors/Node.ts

```diff
@@ -875,13 +876,34 @@ export default class Node extends Executor<NodeEvents, Config, NodePlugins> {
         }
 
         testTask
-          .then(() => {
+          .then(async () => {
             if (!this._sessionSuites) {
               return;
             }
-            return this._loadFunctionalSuites().then(
-              () => (testTask = this._runRemoteTests())
-            );
+
+            let remainingAttempts = 1 + (this.config.functionalRetries || 0);
+            let suites = this._sessionSuites;
+
+            await this._loadFunctionalSuites();
+            while (remainingAttempts && suites.length) {
+              remainingAttempts--;
+              if (suites.length !== this._sessionSuites!.length) {
```

Comment:
It seems odd that `this._sessionSuites` needs a `!` but `suites` doesn't.  Consider adding `const sessionSuites = this._sessionSuites!` and using that to avoid having a bunch of non-null assertions.

Reply:
It doesn't need the `!` any longer. Removed.